app/streaming/session.cpp

```diff
@@ -1119,5 +1119,15 @@ void Session::exec(int displayOriginX, int displayOriginY)
     // When it is complete, it will release our s_ActiveSessionSemaphore
     // reference.
     QThreadPool::globalInstance()->start(new DeferredSessionCleanupTask());
+
+    // Prevents calling of SDL_Quit() in main while cleanup is running by
+    // giving the cleanup time to finish. May occur when all process's windows
+    // are closed at same time.
+    connect(QCoreApplication::instance(), &QCoreApplication::aboutToQuit, []() {
+        if (Session::s_ActiveSessionSemaphore.tryAcquire(1, 10000)) {
```

Comment:
Using `tryAcquire()` means that it's still possible to crash this way if connection cleanup takes longer than 10 seconds.

I don't really like this solution though, because it's a workaround for the real problem which is that we can leave `Session::exec()` with SDL threads still alive (since the `SDL_CloseAudioDevice()` call was taking place as a result of `LiStopConnection()` which happens asynchronously).

I moved SDL audio cleanup onto the main thread in https://github.com/moonlight-stream/moonlight-qt/commit/64bc189010b6550e7da90307a065157a0ecbda89 for unrelated reasons, but it should solve this issue too.

Reply:
Yeah, seems to be fixed indeed 👍